### DISCUSSION TITLE
Building vendure with custom admin ui plugin stopped working.

### DISCUSSION THREAD

**Discussion by agoransson** [vendurehq/vendure#2846] [Q&A] [answered]
With my project, which contains some custom plugins and admin-ui extensions, I recently started getting this error:

```
[server] Usage Error: The nearest package directory (/Users/MyUser/Projects/MyVendureApp/admin-ui) doesn't seem to be part of the project declared in /Users/MyUser/Projects/MyVendureApp.
[server] 
[server] - If /Users/MyUser/Projects/MyVendureApp isn't intended to be a project, remove any yarn.lock and/or package.json file there.
[server] - If /Users/MyUser/Projects/MyVendureApp is intended to be a project, it might be that you forgot to list admin-ui in its workspace configuration.
[server] - Finally, if /Users/MyUser/Projects/MyVendureApp is fine and you intend admin-ui to be treated as a completely separate project (not even a workspace), create an empty yarn.lock file in it.
[server] 
[server] $ yarn run [--inspect] [--inspect-brk] [-T,--top-level] [-B,--binaries-only] [--require #0] <scriptName> ...
[server] npm run dev:server exited with code 0
^C[worker] npm run dev:worker exited with code 0
```

I tried adding an empty `yarn.lock` file but I still get an error, and the project fails to start.

```
[server] Internal Error: vendure-admin-ui@workspace:.: This package doesn't seem to be present in your lockfile; run "yarn install" to update the lockfile
[server]     at Xx.getCandidates (/Users/MyUser/Projects/MyVendureApp/.yarn/releases/yarn-4.1.1.cjs:205:8149)
[server]     at Dd.getCandidates (/Users/MyUser/Projects/MyVendureApp/.yarn/releases/yarn-4.1.1.cjs:141:1311)
[server]     at /Users/MyUser/Projects/MyVendureApp/.yarn/releases/yarn-4.1.1.cjs:210:8409
[server]     at Ky (/Users/MyUser/Projects/MyVendureApp/.yarn/releases/yarn-4.1.1.cjs:140:53916)
[server]     at Fe (/Users/MyUser/Projects/MyVendureApp/.yarn/releases/yarn-4.1.1.cjs:210:8389)
[server]     at async Promise.allSettled (index 0)
[server]     at async Uc (/Users/MyUser/Projects/MyVendureApp/.yarn/releases/yarn-4.1.1.cjs:140:53244)
[server]     at async /Users/MyUser/Projects/MyVendureApp/.yarn/releases/yarn-4.1.1.cjs:210:9140
[server]     at async Qi.startProgressPromise (/Users/MyUser/Projects/MyVendureApp/.yarn/releases/yarn-4.1.1.cjs:140:137284)
[server]     at async Pt.resolveEverything (/Users/MyUser/Projects/MyVendureApp/.yarn/releases/yarn-4.1.1.cjs:210:7138)
```

I'm unsure why this started failing all of a sudden; has anyone else had this problem? Also, I'm using NPM - so why is yarn involved? Is that used underneath somewhere in the process of building the admin ui?

I have tried moving to a new project, and moving all my plugins there - and I still get the same issue.

I'm using Vendure 2.2.2. And currently building it on Mac.

Thanks

**Comment by agoransson** [comment]
Another attempt, with a fresh install of vendure, I managed to get this working again. The only difference is I added each plugin one-by-one, and ensuring the app starts up between. Seems that made the trick - still unsure why it started failing before.

**Comment by agoransson** [comment, marked as answer]
I got this again now; seems it might be related to node version. When I set node to use v 20 instead of 21 it stopped complaining.

`nvm use 20`

**Reply by michaelbromley** [reply]
Very possibly. Odd-numbered node releases are not officially supported by Vendure. There is a reason for this - odd versions often introduce changes that break things and then get stabilized in the next even version. 
![image](https://github.com/vendure-ecommerce/vendure/assets/6275952/e24805eb-8d2b-4bca-870d-488ff3b9ee0a)
